applications/OptimizationApplication/python_scripts/controls/master_control.py

```diff
@@ -162,23 +163,34 @@ def MapGradient(self, physical_space_gradient_variable_and_collective_expression
         return mapped_gradients
 
     @time_decorator()
-    def Update(self, update_collective_expressions: KratosOA.CollectiveExpression) -> 'dict[Control, bool]':
+    def Update(self, updated_control_vars: KratosOA.CollectiveExpression or numpy.ndarray) -> 'dict[Control, bool]':
         """Update each control with given collective expression's respective container expression.
 
         Args:
-            update_collective_expressions (KratosOA.CollectiveExpression): Update
+            updated_control_vars (KratosOA.CollectiveExpression or numpy.ndarray): Update
 
         Raises:
             RuntimeError: If number of controls and number of container expressions mismatch.
 
         Returns:
             dict[Control, bool]: A map with control and a boolean whether the update changed anything in that control.
         """
-        if len(self.__list_of_controls) != len(update_collective_expressions.GetContainerExpressions()):
+
+        if isinstance(updated_control_vars, numpy.ndarray):
```

Comment:
I am not sure in this implementation. Why not to convert numpy back to container expression and keep standart well tested routine?

Reply:
@Igarizza that's a very good point. Applied !